src/main/java/net/devtech/arrp/impl/RuntimeResourcePackImpl.java

```diff
@@ -418,27 +420,28 @@ public Identifier getId() {
 	/**
 	 * pack.png and that's about it I think/hope
 	 *
-	 * @param fileName the name of the file, can't be a path tho
+	 * @param segments the name of the file, can't be a path tho
 	 * @return the pack.png image as a stream
 	 */
 	@Override
-	public InputStream openRoot(String fileName) {
-		if(!fileName.contains("/") && !fileName.contains("\\")) {
+	public InputSupplier<InputStream> openRoot(String... segments) {
+		if(segments.length == 1) {
+			String fileName = segments[0];
 			this.lock();
 			Supplier<byte[]> supplier = this.root.get(fileName);
```

Comment:
I think instead we should change the root hashmap to support the new string array system, I would do `Map<List<String>, Supplier<byte[]>>` and wrap the incomming `String... segments` into a `Arrays.asList()`, I'm assuming `segments` is just a `string.split("/")` on the original path, so we can maintain backwards compat by making the addRoot resource do a `string.split("/")` before storing it in the map

Reply:
I probably fixed it.